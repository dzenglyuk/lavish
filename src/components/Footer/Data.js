export const infoLinks = [
    {
        title: "About Us",
        links: [
            {
                name: "How it works",
                value: "/sign-up",
            },
            {
                name: "Testimonials",
                value: "/",
            },
            {
                name: "Careers",
                value: "/",
            },
            {
                name: "Investors",
                value: "/",
            },
            {
                name: "Terms of Service",
                value: "/",
            },
        ],
    },
    {
        title: "Contact Us",
        links: [
            {
                name: "Contact",
                value: "/",
            },
            {
                name: "Support",
                value: "/",
            },
            {
                name: "Destinations",
                value: "/",
            },
            {
                name: "Sponsorships",
                value: "/",
            },
        ],
    },
];

export const socialLinks = [
    {
        title: "Videos",
        links: [
            {
                name: "Submit Video",
                value: "/",
            },
            {
                name: "Ambassadors",
                value: "/",
            },
            {
                name: "Agency",
                value: "/",
            },
            {
                name: "Influencer",
                value: "/",
            },
        ],
    },
    {
        title: "Social Media",
        links: [
            {
                name: "Instagram",
                value: "/",
            },
            {
                name: "Facebook",
                value: "/",
            },
            {
                name: "Youtube",
                value: "/",
            },
            {
                name: "Twitter",
                value: "/",
            },
        ],
    },
];

export const subscription = {
    heading: "Join our exclusive membership to receive the latest news and trends",
    text: "You can unsubscribe at any time."
};