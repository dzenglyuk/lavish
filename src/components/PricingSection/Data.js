const starterAccount = {
    info: {
        name: "Starter",
        price: "$8.99",
        period: "per month"
    },
    features: ["100 Transactions", "2% Cash Back", "$10,000 Limit"]
};

const goldAccount = {
    info: {
        name: "Gold",
        price: "$29.99",
        period: "per month"
    },
    features: ["1000 Transactions", "3.5% Cash Back", "$100,000 Limit"]
};

const diamondAccount = {
    info: {
        name: "Diamond",
        price: "$99.99",
        period: "per month"
    },
    features: ["Unlimited Transactions", "5% Cash Back", "Unlimited Spending"]
};

export {
    starterAccount,
    goldAccount,
    diamondAccount
};