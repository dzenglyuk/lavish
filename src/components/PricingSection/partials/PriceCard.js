import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import Button from "../../Button/Main";

const PriceCard = ({ path, info, features, icon, btnColor }) => {
    const { name, price, period } = info;
    return (
        <Link to={path} className="pricing__container-card">
            <div className="pricing__container-cardInfo">
                <div className="icon">{icon}</div>
                <h3> {name} </h3>
                <h4> {price} </h4>
                <p> {period} </p>
                <ul className="pricing__container-features">
                    {features.map((feature) => (
                        <li key={feature}>{feature}</li>
                    ))}
                </ul>
                <Button btnSize="btn--wide" btnColor={btnColor}>
                    Choose Plan
                </Button>
            </div>
        </Link>
    );
};

PriceCard.defaultProps = {
    path: "/sign-up",
    btnColor: "primary",
};

PriceCard.propTypes = {
    icon: PropTypes.element.isRequired,
    features: PropTypes.arrayOf(PropTypes.string).isRequired,
    info: (props, propName, componentName) => {
        const propValue = (props || {})[propName];
        const requiredProps = ["name", "price", "period"];

        if (propValue === undefined)
            return new Error(
                `${propName} of component ${componentName} is required`
            );

        if (typeof propValue !== "object")
            return new Error(
                `${propName} of component ${componentName} must be array`
            );

        requiredProps.map((prop) =>
            prop in propValue
                ? null
                : new Error(
                      `${propName} of component ${componentName} must include ${prop} value`
                  )
        );

        return null;
    },
};

export default PriceCard;
