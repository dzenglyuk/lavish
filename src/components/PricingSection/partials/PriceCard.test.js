import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { StaticRouter as Router } from "react-router";
import { starterAccount, goldAccount, diamondAccount } from "../Data";
import { FaFire } from "react-icons/fa";

import PriceCard from "./PriceCard";

test("Renders correctly", () => {
    const { getByText } = render(
        <Router>
            <PriceCard {...starterAccount} icon={<FaFire />}/>
        </Router>
    );
    getByText("Starter");
    getByText("$8.99");
    getByText("100 Transactions");
    getByText("2% Cash Back");
    getByText("$10,000 Limit");
    getByText("Choose Plan");
});

// test("Throws an error on required prop", () => {
//     const propObj = { features: starterAccount.features };
//     console.error = jest.fn();
//     const { getByText } = render(
//         <Router>
//             <PriceCard {...propObj} icon={<FaFire />}/>
//         </Router>
//     );
//     expect(console.error).toHaveBeenCalledTimes(1);
// });

test("Matches the snapshot 1", () => {
    const tree = renderer
        .create(
            <Router>
                <PriceCard {...starterAccount} icon={<FaFire />}/>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("Matches the snapshot 2", () => {
    const tree = renderer
        .create(
            <Router>
                <PriceCard {...goldAccount} icon={<FaFire />}/>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("Matches the snapshot 3", () => {
    const tree = renderer
        .create(
            <Router>
                <PriceCard {...diamondAccount} icon={<FaFire />}/>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
