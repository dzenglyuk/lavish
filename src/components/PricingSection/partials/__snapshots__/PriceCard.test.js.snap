// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1 1`] = `
<a
  className="pricing__container-card"
  href="/sign-up"
  onClick={[Function]}
>
  <div
    className="pricing__container-cardInfo"
  >
    <div
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 384 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216 23.86c0-23.8-30.65-32.77-44.15-13.04C48 191.85 224 200 224 288c0 35.63-29.11 64.46-64.85 63.99-35.17-.45-63.15-29.77-63.15-64.94v-85.51c0-21.7-26.47-32.23-41.43-16.5C27.8 213.16 0 261.33 0 320c0 105.87 86.13 192 192 192s192-86.13 192-192c0-170.29-168-193-168-296.14z"
        />
      </svg>
    </div>
    <h3>
       
      Starter
       
    </h3>
    <h4>
       
      $8.99
       
    </h4>
    <p>
       
      per month
       
    </p>
    <ul
      className="pricing__container-features"
    >
      <li>
        100 Transactions
      </li>
      <li>
        2% Cash Back
      </li>
      <li>
        $10,000 Limit
      </li>
    </ul>
    <button
      className="btn 
			btn--primary
			btn--wide
			primary"
    >
      Choose Plan
    </button>
  </div>
</a>
`;

exports[`Matches the snapshot 2 1`] = `
<a
  className="pricing__container-card"
  href="/sign-up"
  onClick={[Function]}
>
  <div
    className="pricing__container-cardInfo"
  >
    <div
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 384 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216 23.86c0-23.8-30.65-32.77-44.15-13.04C48 191.85 224 200 224 288c0 35.63-29.11 64.46-64.85 63.99-35.17-.45-63.15-29.77-63.15-64.94v-85.51c0-21.7-26.47-32.23-41.43-16.5C27.8 213.16 0 261.33 0 320c0 105.87 86.13 192 192 192s192-86.13 192-192c0-170.29-168-193-168-296.14z"
        />
      </svg>
    </div>
    <h3>
       
      Gold
       
    </h3>
    <h4>
       
      $29.99
       
    </h4>
    <p>
       
      per month
       
    </p>
    <ul
      className="pricing__container-features"
    >
      <li>
        1000 Transactions
      </li>
      <li>
        3.5% Cash Back
      </li>
      <li>
        $100,000 Limit
      </li>
    </ul>
    <button
      className="btn 
			btn--primary
			btn--wide
			primary"
    >
      Choose Plan
    </button>
  </div>
</a>
`;

exports[`Matches the snapshot 3 1`] = `
<a
  className="pricing__container-card"
  href="/sign-up"
  onClick={[Function]}
>
  <div
    className="pricing__container-cardInfo"
  >
    <div
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 384 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216 23.86c0-23.8-30.65-32.77-44.15-13.04C48 191.85 224 200 224 288c0 35.63-29.11 64.46-64.85 63.99-35.17-.45-63.15-29.77-63.15-64.94v-85.51c0-21.7-26.47-32.23-41.43-16.5C27.8 213.16 0 261.33 0 320c0 105.87 86.13 192 192 192s192-86.13 192-192c0-170.29-168-193-168-296.14z"
        />
      </svg>
    </div>
    <h3>
       
      Diamond
       
    </h3>
    <h4>
       
      $99.99
       
    </h4>
    <p>
       
      per month
       
    </p>
    <ul
      className="pricing__container-features"
    >
      <li>
        Unlimited Transactions
      </li>
      <li>
        5% Cash Back
      </li>
      <li>
        Unlimited Spending
      </li>
    </ul>
    <button
      className="btn 
			btn--primary
			btn--wide
			primary"
    >
      Choose Plan
    </button>
  </div>
</a>
`;
